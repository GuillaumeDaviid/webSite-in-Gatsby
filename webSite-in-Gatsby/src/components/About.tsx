import * as React from 'react'
import './styles/About.scss'

export default function About() {
    return(
        <section id="introduction">
        <div className="About">

            <h2 className="About_title" data-testid="title">A PROPOS</h2>
            <div className="About_separator" data-testid="separator"></div>
            <p className="About_text" data-testid="text">Développeur web <strong>Front-End</strong> Freelance dans les Hauts-de-France (Lille), 
                Spécialisé <strong>Javascript / React.js</strong> et dans l&apos;intégration /
                développement responsive <strong>HTML/CSS/JS</strong>. Je vous accompagne dans le développement de votre projet web. Vous retrouverez ici mon portfolio et un moyen de me contacter.</p>

                <div className="creativity" data-testid="creativity">
                    <div className='creativity_block'>

                    
                    <a className='creativity_btn' target="_blank">Créativité</a>
        
                </div>
                </div>
                
        </div>
        </section>
    )
}