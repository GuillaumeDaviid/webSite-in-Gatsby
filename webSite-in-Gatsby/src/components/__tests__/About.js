import React from "react"
import renderer from "react-test-renderer"
import About from "../About.tsx"
import { render, screen } from "@testing-library/react"


describe("About", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<About />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("should have a title element", () => {
        render(<About />)
    
        const title = screen.getByTestId('title');
        expect(title).toBeTruthy();
    })

    it("should have a separator element", () => {
        render(<About />)
    
        const separator = screen.getByTestId('separator');
        expect(separator).toBeTruthy();
    })

    it("should have a text element", () => {
        render(<About />)
    
        const text = screen.getByTestId('text');
        expect(text).toBeTruthy();
    })

    it("should have a creativity block", () => {
        render(<About />)
    
        const creativity = screen.getByTestId('creativity');
        expect(creativity).toBeTruthy();
    })
})