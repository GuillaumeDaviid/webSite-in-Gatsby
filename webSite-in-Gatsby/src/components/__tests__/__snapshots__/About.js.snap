// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<section
  id="introduction"
>
  <div
    className="About"
  >
    <h2
      className="About_title"
      data-testid="title"
    >
      A PROPOS
    </h2>
    <div
      className="About_separator"
      data-testid="separator"
    />
    <p
      className="About_text"
      data-testid="text"
    >
      Développeur web 
      <strong>
        Front-End
      </strong>
       Freelance dans les Hauts-de-France (Lille), Spécialisé 
      <strong>
        Javascript / React.js / NextJS / Gatsby
      </strong>
       et dans l'intégration / développement responsive 
      <strong>
        HTML/CSS/JS
      </strong>
      . Je vous accompagne dans le développement de votre projet web. Vous retrouverez ici mon portfolio et un moyen de me contacter.
    </p>
    <a
      className="btn_link"
      href="#contact"
    >
      <button
        className="btn btn_contact"
      >
        ME CONTACTER
      </button>
    </a>
    <div
      className="creativity"
      data-testid="creativity"
    >
      <div
        className="creativity_block"
      >
        <a
          className="creativity_btn"
          target="_blank"
        >
          Créativité
        </a>
      </div>
    </div>
  </div>
</section>
`;
